Automattic/pocket-casts-android
# Make PodcastRating::average nullable

## Description
It feels odd to me that our `PodcastRatings` object has a nullable `total` field and a non-null `average` field. I feel like they should either both be nullable or both not be nullable. 

This PR originally updated the `total` field so it is not nullable, and if we didn't get a `total` value back from the backend, it used a `total` of 0, which is what we were already doing elsewhere in the app. I originally went with making them non-null just because we already treat any null values as `0` in the app, so we didn't really need to preserve the null values.

After PR review feedback (https://github.com/Automattic/pocket-casts-android/pull/1454#issuecomment-1756759270) though, I've adopted the alternative approach of making both `average` and `total` nullable. The benefit of this is that the `total` and `average` fields are not required parts of the ratings API response, so this makes our database consistent with that.

This does add in a non-trivial database migration, so if you think adding the migration is too much of a cost to just remove a null field, let me know.

> **Note**
> This PR builds on https://github.com/Automattic/pocket-casts-android/pull/1453

## Testing Instructions
1. Install the app from `main`
2. Open the podcast page for a few podcasts including one with no ratings like "The Big Dig"
3. Install the app from this PR
4. Open the podcast pages for the same podcasts you opened before along with a few new ones and verify that the ratings are displayed.

## Checklist
